shuup/admin/modules/users/views/detail.py

```diff
@@ -223,3 +243,38 @@ def post(self, request, *args, **kwargs):
     def dispatch(self, request, *args, **kwargs):
         self.model = get_user_model()
         return super(UserDetailView, self).dispatch(request, *args, **kwargs)
+
+
+class LoginAsUserView(DetailView):
+    model = get_user_model()
+
+    def post(self, request, *args, **kwargs):
+        front_url = get_front_url()
+        user = self.get_object()
+        username_field = self.model.USERNAME_FIELD
+        person_contact = PersonContact.objects.filter(user=user).first()
+
+        if not front_url:
+            raise Problem(_("No shop configured."))
+        if not person_contact:
+            raise Problem(_("User is not bound to a person contact."))
```

Comment:
Is this 100% necessary? I know the shuup front middleware creates a contact if one doesn't exist for a user, although not sure if that applies in this case since we're logging in through admin.